relay-server/src/services/store.rs

```diff
@@ -989,21 +994,86 @@ impl StoreService {
+                received % 1_000_000_000
+            } else {
+                0
+            };
+            let mut trace_item = TraceItem {
+                item_type: TraceItemType::Log.into(),
+                organization_id: scoping.organization_id.value(),
+                project_id: scoping.project_id.value(),
+                received: Some(Timestamp {
+                    seconds: safe_timestamp(received_at) as i64,
+                    nanos: received_nanos as i32,
+                }),
+                retention_days: retention_days.into(),
+                timestamp: Some(Timestamp {
+                    seconds: timestamp_seconds as i64,
+                    nanos: timestamp_nanos as i32,
+                }),
+                trace_id: log.trace_id.to_string(),
+                item_id,
+                attributes: Default::default(),
+                client_sample_rate: 1.0,
+                server_sample_rate: 1.0,
+            };
+
+            for (name, attribute) in log.attributes.unwrap_or_default() {
+                if let Some(attribute_value) = attribute {
+                    if let Some(v) = attribute_value.value {
+                        let any_value = match v {
+                            LogAttributeValue::String(value) => AnyValue {
+                                value: Some(Value::StringValue(value)),
+                            },
+                            LogAttributeValue::Int(value) => AnyValue {
+                                value: Some(Value::IntValue(value)),
+                            },
+                            LogAttributeValue::Bool(value) => AnyValue {
+                                value: Some(Value::BoolValue(value)),
+                            },
+                            LogAttributeValue::Double(value) => AnyValue {
+                                value: Some(Value::DoubleValue(value)),
+                            },
+                            LogAttributeValue::Unknown(_) => continue,
+                        };
+
+                        trace_item.attributes.insert(name.into(), any_value);
+                    }
+                }
+            }
+
+            let mut message = Vec::new();
+
+            if trace_item.encode(&mut message).is_err() {
+                return Err(StoreError::ProtobufEncodingFailed);
+            }
 
             let message = KafkaMessage::Log {
-                headers: BTreeMap::from([(
-                    "project_id".to_string(),
-                    scoping.project_id.to_string(),
-                )]),
-                message: log,
+                headers: BTreeMap::from([
+                    ("project_id".to_string(), scoping.project_id.to_string()),
+                    (
+                        "item_type".to_string(),
+                        TraceItemType::Log.as_str_name().to_string(),
+                    ),
+                ]),
```

Comment:
```suggestion
                headers: BTreeMap::from([
                    ("project_id".to_owned(), scoping.project_id.to_owned()),
                    (
                        "item_type".to_owned(),
                        TraceItemType::Log.as_str_name().to_owned(),
                    ),
                ]),
```

Reply:
`scoping.project_id` is a `ProjectId` so `to_owned()` won't work, I have to use `to_string()`.